gobblin-core/src/main/java/gobblin/source/extractor/filebased/FileBasedExtractor.java

```diff
@@ -94,6 +109,7 @@ public D readRecord(@Deprecated D reuse) throws DataRecordException, IOException
       if (!filesToPull.isEmpty()) {
         currentFile = filesToPull.remove(0);
         currentFileItr = downloadFile(currentFile);
+        incrementBytesReadCounter();
```

Comment:
This approach seems to be incrementing the counter before it downloads the file, correct? It may be better to increment the counter after the file has been downloaded. I think it makes the metrics more accurate. For example, if the download of a file fails, the metrics will still report that the whole file was downloaded (even though it wasn't).